grammar.js

```diff
@@ -293,16 +293,16 @@ module.exports = grammar({
     ),
 
     generic_type: $ => seq(
-      field("type", $._type_identifier),
-      field("type_arguments", $.type_arguments),
+      field('type', $._type_identifier),
+      field('type_arguments', $.type_arguments),
     ),
 
-    type_arguments: $ => seq(
+    type_arguments: $ => prec.dynamic(2, seq(
```

Comment:
maybe a comment would be useful here to explain the need for prec.dynamic, that is to prefer call_expression to index_expression, and why this works with the value 2.

Reply:
I think it is because `_type_identifier` rule in `_simple_type` rule has its dynamic precedence -1. For example, if `type_arguments` rule had 1 dynamic precedence typical syntax tree like `(type_arguments (_simple_type (type_identifier)))` will have 0 dynamic precedence so it makes no difference between `index_expression`.
Should I open a pull request to write that comment?

Reply:
I think those comments are useful for the next person that sees those prec.dynamic(2)
and wonder what it is.
Conflicts are hard to understand, so the more we have comments like this the better I think.